ApeSwapFinance/frontend
# [FEAT] - LHD Historical chart

## Depends on: 
- ~https://github.com/ApeSwapFinance/lhd-api/pull/144~ https://github.com/ApeSwapFinance/lhd-api/pull/147
- Liquidity debt data pending

## ⛑️ **What was done:**
- Added Historical Chart to LHD Token profile
- Added Tab Navigation to  LHD Token profile
- Made `TabNavigation` agnostic and moved it  to Components folder
- Created custom hook to cache and fetch data

### 🧪 **How to test:**
- Visit `/liquidity-health`
- Click on any token
- 'Strength' and 'Liquidity' tab navigation should be visible
- Should **NOT** SSR to improve perceived loading speed
- Clicking on 'Historic' should display historical chart
- If loading, display loading spinner
- If fully loaded, display Line chart with one line toggle
- Checkboxes for lines should be displayed at the bottom of the chart
- Clicking on checkboxes should toggle to correct line on the chart
- At lease one checkbox must be toggled at any given time
- $ value scales should be displayed on the Y axis on the left side of the chart properly formatted
- Score value scales should be displayed on the Y axis on the right side of the chart
- Hovering on the line chart should display a tooltip with the amounts correctly formatted
- Y axis labels should no be displayed on mobile

## **Demo:**
- Desktop

https://github.com/ApeSwapFinance/frontend/assets/47801291/d5dc9a3f-4320-458d-b58e-e76b74abfaf4

- Mobile

https://github.com/ApeSwapFinance/frontend/assets/47801291/647c305e-fd64-4893-aba0-fe4e3f2e745d
